core/types/transaction_signing.go

```diff
@@ -219,7 +219,7 @@ func (s pragueSigner) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big
 	}
 	// Check that chain ID of tx matches the signer. We also accept ID zero here,
 	// because it indicates that the chain ID was not specified in the tx.
-	if txdata.ChainID != 0 && new(big.Int).SetUint64(txdata.ChainID).Cmp(s.chainId) != 0 {
+	if txdata.ChainID != nil && txdata.ChainID.CmpBig(s.chainId) != 0 {
```

Comment:
is this correct?

should it be `if txdata.ChainID.Sign() != 0 && txdata.ChainID.ToBig().Cmp(s.chainId) != 0 {` ?

I think it was copied from the `cancunSigner`, but it's not same with the `cancunSigner`

Reply:
a thinking about it, It is correct.

I have made a PR #31032 to update the cancunSigner to align with it